spotbugs/src/main/java/edu/umd/cs/findbugs/PriorityAdjustment.java

```diff
@@ -0,0 +1,53 @@
+package edu.umd.cs.findbugs;
+
+import java.util.regex.Pattern;
+
+class PriorityAdjustment {
+    private static final Pattern NUMBER_PATTERN = Pattern.compile("\\d+");
+    private final int delta;
+    private final boolean isDelta;
+
+    PriorityAdjustment(String value) {
+        if (value == null || value.isEmpty()) {
+            throw new IllegalArgumentException("Priority adjustment value cannot be null or empty");
+        }
+        if (value.startsWith("+")) {
+            String num = value.substring(1);
+            if (!NUMBER_PATTERN.matcher(num).matches()) {
+                throw wrongFormat(value);
+            }
+            isDelta = true;
+            delta = Integer.parseInt(num);
+        } else if (value.startsWith("-")) {
+            String num = value.substring(1);
+            if (!NUMBER_PATTERN.matcher(num).matches()) {
+                throw wrongFormat(value);
+            }
+            isDelta = true;
+            delta = -Integer.parseInt(num);
+        } else if (NUMBER_PATTERN.matcher(value).matches()) {
+            isDelta = false;
+            delta = Integer.parseInt(value);
+        } else if ("raise".equals(value)) {
+            isDelta = true;
+            delta = -1;
+        } else if ("lower".equals(value)) {
+            isDelta = true;
+            delta = 1;
+        } else if ("suppress".equals(value)) {
+            isDelta = true;
+            delta = +100;
+        } else {
+            throw wrongFormat(value);
+        }
+    }
+
+    private IllegalArgumentException wrongFormat(String value) {
+        return new IllegalArgumentException("Invalid priority adjustment value: " + value
+                + ". Must be +[integer], -[integer], [integer] or one of raise|lower|suppress");
+    }
+
+    public int adjust(int priority) {
+        return isDelta ? priority + delta : delta;
```

Comment:
After reading this method I would propose to rename `delta` field to `adjustmentValue`.